java/fury-core/src/main/java/org/apache/fury/Fury.java

```diff
@@ -1220,6 +1239,43 @@ public Object deserializeJavaObjectAndClass(FuryReadableChannel channel) {
     return deserializeJavaObjectAndClass(buf);
   }
 
+  @Override
+  public <T> T copy(T obj) {
+    if (obj == null) {
+      return null;
+    }
+    copyDepth++;
+    try {
+      return (T) classResolver.getOrUpdateClassInfo(obj.getClass()).getSerializer().copy(obj);
```

Comment:
Would be better to use a switch here to create a fastpath for common types, which can avoid the virtual method call cost